fdbserver/CommitProxyServer.actor.cpp

```diff
@@ -1165,10 +1169,19 @@ void writeMutation(CommitBatchContext* self, int64_t tenantId, const MutationRef
 		                   self->pProxyCommitData->db->get().client.tenantMode == TenantMode::REQUIRED;
 		CODE_PROBE(isRawAccess, "Raw access to tenant key space");
 		self->toCommit.writeTypedMessage(mutation);
+		return mutation;
 	} else {
-		Arena arena;
-		self->toCommit.writeTypedMessage(
-		    mutation.encrypt(self->cipherKeys, tenantId /*domainId*/, arena, BlobCipherMetrics::TLOG));
+		MutationRef encryptedMutation;
+		if (encryptedMutationOpt.present()) {
+			CODE_PROBE(true, "using already encrypted mutation");
+			ASSERT(encryptedMutation.isEncrypted());
```

Comment:
Shall we add a simulation only check which compare if the encrypted mutation is same as the original mutation? Though we will need to ensure the cipher key is available beforehand.

Reply:
Yeah good point i think this would be a useful check in simulation

Reply:
I like this idea to extend it further; checks can be of different flavor. For instance:
1. Just decrypt and compare checksum (maybe)
2. Fetch encryption keys and validate key-sanity along with decrypt and compare. Validate EncryptionHeader construction

Randomly choose the validation effort would be good.